### PR TITLE
SNAP-956 Refactor Snapshot Search UI

### PR DESCRIPTION
**Pull request by ggomez-cambriasolutions**
### Jira Story

https://osi-cwds.atlassian.net/browse/SNAP-956

## Description
Refactor Search UI to reflect the newest design.

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/assets/stylesheets/_colors.scss [hunk 1]
```diff
@@ -12,6 +12,7 @@ $dark-blue: #053d53;
 $light-green: #3fbfad;
 $medium-green: #2b8377;
 $dark-green: #065144;
+$green: rgba(51, 120, 109, 1);
```
[the remark this answers is not in the document]
Comment (ggomez-cambriasolutions) [reply app/assets/stylesheets/_colors.scss]: Yes. The design is posted in the story, if you'd like to take a look.

File: app/javascript/common/search/age/ApproximateAgeNumberSelect.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import AgeNumberSelect from 'common/search/age/AgeNumberSelect'
+
+class ApproximateAgeNumberSelect extends React.Component {
+  render() {
+    const {onChange, ...rest} = this.props
```
[the remark this answers is not in the document]
Comment (ggomez-cambriasolutions) [reply app/javascript/common/search/age/ApproximateAgeNumberSelect.jsx]: it's a pattern that was already in another select component. so I followed that style.

File: app/javascript/common/search/PersonSearchButtonGroup.jsx [hunk 3]
```diff
@@ -29,17 +29,17 @@ const PersonSearchButtonGroup = ({
...
       >
         Search
       </button>
       <button
-        className="btn btn-primary person-search-button cancel"
+        className="btn person-search-button clear"
         onClick={onCancel}
       >
-        Cancel
+        Clear All
```
[the remark this answers is not in the document]
Comment (ggomez-cambriasolutions) [reply app/javascript/common/search/PersonSearchButtonGroup.jsx]: Good call. I placed a css override to ensure the text is not transformed to uppercase.

File: app/javascript/common/search/age/ApproximateAgeNumberSelect.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ggomez-cambriasolutions) [reply app/javascript/common/search/age/ApproximateAgeNumberSelect.jsx]: On second thought, you have a valid point. I'm going to update the component.

**Comment by ggomez-cambriasolutions** [comment]
@pokharelmanish We disabled those element because they aren't functional. The functionality will be implemented in a future story. This is consistent with the previous iteration of the search UI, where we disabled fields that were not functional. I'll follow up with Sri to decide the best way to go forward.

**Comment by pokharelmanish** [comment]
Thanks for replying my comment. 
@ggomez-cambriasolutions It would be great if you could attach the screenshot after making the changes (It will be easier for the reviewer to review because it is a visual change).

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please

**Comment by ggomez-cambriasolutions** [comment]
Screen Shot 2019-03-12 at 5 02 38 PM

**Comment by ggomez-cambriasolutions** [comment]
Note that I adjusted some of the elements to better align with the grid. This is based on a request from QA and PO, even though the design has elements that are not aligned very well with the grid.

**Comment by crystalctaylor** [comment]
> In design, I see `Approximate Age`, `Months`, `Years` and `Number` is enabled but I see it is disabled here. I think `Search` title should be `Snapshot Search`.
> And few questions, other than that looks good to me.

For this issue, yes, the Approx Age section should be disabled on the page until the following future story is played: https://osi-cwds.atlassian.net/browse/SNAP-894
This is because PO agreed in Slack that all fields not currently part of search be disabled on the UI until that specific functionality is implemented.

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please

**Comment by ggomez-cambriasolutions** [comment]
jenkins retest this please

**Pull request merged**